front/app/containers/Admin/projects/project/timeline/edit.tsx

```diff
@@ -236,33 +239,40 @@ class AdminProjectTimelineEdit extends PureComponent<
 
   handlePhaseFileOnAdd = (newFile: UploadFile) => {
     this.setState((prevState) => {
-      const isDuplicate = prevState.phaseFiles.some(
-        (file) => file.base64 === newFile.base64
+      const modifiedNewFile = {
+        name: newFile.name || newFile.filename,
+        size: newFile.size,
+        remote: false,
+        base64: newFile.base64,
+      };
+
+      const isDuplicate = prevState.inStatePhaseFiles.some(
+        (file) => file.name === newFile.name
```

Comment:
I see the previous implementation checked the base64 to determine if a file was a duplicate - did you replace it with file name to speed up the upload process? Or was there another reason?

Reply:
Yes, we won't always have a base64. Not from the already uploaded files. That would require us downloading them

Reply:
Ahh that's right. Ok, makes sense to me :) Thanks!

Reply:
Actually, let me make it better and check for base64 for local files and name for remote ones or when the base64 is not present.